src/Orchard.Web/Modules/Orchard.Fields/Drivers/LinkFieldDriver.cs

```diff
@@ -57,11 +57,30 @@ protected override DriverResult Editor(ContentPart part, LinkField field, IUpdat
                 if (settings.Required && String.IsNullOrWhiteSpace(field.Value)) {
                     updater.AddModelError(GetPrefix(field, part), T("Url is required for {0}", T(field.DisplayName)));
                 }
-                else if (!String.IsNullOrWhiteSpace(field.Value) && !Uri.IsWellFormedUriString(field.Value, UriKind.RelativeOrAbsolute)) {
-                    updater.AddModelError(GetPrefix(field, part), T("{0} is an invalid url.", field.Value));
-                }
                 else if (settings.LinkTextMode == LinkTextMode.Required && String.IsNullOrWhiteSpace(field.Text)) {
                     updater.AddModelError(GetPrefix(field, part), T("Text is required for {0}.", T(field.DisplayName)));
+                } else if (!String.IsNullOrWhiteSpace(field.Value)) {
+                    // Check if it's a valid uri, considering that there may be the link to an anchor only
+                    // e.g.: field.Value = "#divId"
+                    // Take everything before the first "#" character and check if it's a valid uri.
+                    // If there is no character before the first "#", consider the value as a valid one (because it is a reference to a div inside the same page)
+                    var index = field.Value.IndexOf('#');
```

Comment:
A technique I like to use for these cases is `field.Value.Split('#', 2)`. It will only give you an array or 1 or 2 values even if there are more `#`. This way you have access to both parts directly.

Reply:
As suggested, refactored the code to use string split function.